packages/mobile/src/components/scrubber/usePosition.ts

```diff
@@ -0,0 +1,57 @@
+import { useCallback, useEffect, useRef } from 'react'
+
+import { formatSeconds } from '@audius/common'
+import type { TextInput } from 'react-native'
+
+export const usePosition = (
+  mediaKey: string,
+  duration: number,
+  isPlaying: boolean,
+  isInteracting: boolean
+) => {
+  const positionRef = useRef(0)
+  const positionElementRef = useRef<TextInput>(null)
+
+  const setPosition = useCallback((position: number) => {
+    positionRef.current = position
+    positionElementRef.current?.setNativeProps({
+      text: formatSeconds(positionRef.current)
+    })
+  }, [])
+
+  useEffect(() => {
+    let outerTimeout: NodeJS.Timeout
+    let currentTimeout: NodeJS.Timeout
+
+    const updatePosition = () => {
+      const timeout = setTimeout(() => {
+        positionRef.current += 1
+
+        positionElementRef.current?.setNativeProps({
+          text: formatSeconds(positionRef.current)
+        })
+
+        if (positionRef.current <= duration) {
+          currentTimeout = updatePosition()
```

Comment:
Could you maybe add a comment about how outerTimeout and currentTimeout interact, it's kinda confusing

Reply:
yeah totally agreed